src/coreclr/vm/threads.cpp

```diff
@@ -2517,6 +2525,13 @@ Thread::~Thread()
 
     // Wait for another thread to leave its loop in DeadlockAwareLock::TryBeginEnterLock
     CrstHolder lock(&g_DeadlockAwareCrst);
```

Comment:
This takes a lock till the end of the method. I do not think you meant to execute this under that lock.